lib/bolt/puppetfile.rb

```diff
@@ -0,0 +1,231 @@
+# frozen_string_literal: true
+
+require 'pathname'
+
+require 'bolt/error'
+require 'bolt/puppetfile/module'
+
+# This class is used to interact with a Puppetfile. It includes methods for
+# parsing an existing Puppetfile, writing a new Puppetfile, resolving module
+# dependencies, and installing modules from a Puppetfile.
+#
+module Bolt
+  class Puppetfile
+    attr_reader :modulepath, :modules, :path
+
+    def initialize(path:, config:, modulepath:, modules: [])
+      @path       = path
+      @config     = config
+      @modulepath = modulepath
```

Comment:
It looks like `modulepath` can just be an argument to `install`? In that case, it ought to be a single directory and not a modulepath, since we only care about a single directory.